akka-stream/src/main/scala/akka/stream/javadsl/Sink.scala

```diff
@@ -235,11 +236,26 @@ object Sink {
    * of the actor will grow. For potentially slow consumer actors it is recommended
    * to use a bounded mailbox with zero `mailbox-push-timeout-time` or use a rate
    * limiting operator in front of this `Sink`.
-   *
    */
   def actorRef[In](ref: ActorRef, onCompleteMessage: Any): Sink[In, NotUsed] =
     new Sink(scaladsl.Sink.actorRef[In](ref, onCompleteMessage))
 
+  /**
+   * Sends the elements of the stream to the given `ActorRef`.
+   * If the target actor terminates the stream will be canceled.
+   * When the stream is completed successfully a [[Status.Success]] (with status `Done`) message will be sent to the destination actor.
+   * When the stream is completed with failure a [[akka.actor.Status.Failure]] message will be sent to the destination actor.
+   *
+   * It will request at most `maxInputBufferSize` number of elements from
+   * upstream, but there is no back-pressure signal from the destination actor,
+   * i.e. if the actor is not consuming the messages fast enough the mailbox
+   * of the actor will grow. For potentially slow consumer actors it is recommended
+   * to use a bounded mailbox with zero `mailbox-push-timeout-time` or use a rate
+   * limiting operator in front of this `Sink`.
+   */
+  def actorRef[In](ref: ActorRef): Sink[In, NotUsed] =
```

Comment:
I'm not sure I like this addition. The `Status` messages are very vague. I would prefer to go in the opposite direction, to only promote the one with explicit `onCompleteMessage` and `onFailureMessage` parameter and deprecate the others.

wdyt @johanandren @raboof ?

Reply:
I tried to mirror the `Source.actorRef` API (relying on `Status` messages). In that case also open the API on the source to be able to provide the `completionMatcher` and `failureMatcher`? (akka typed one already does)

Reply:
Sorry, didn't notice this question from Patrik 25 days ago. I'm ok with this, but I also don't think it is super valuable since it is so little code to get the same semantics with the explicit completion/failure factory.

Reply:
@patriknw should we got the 'typed' api way accepting completionMatcher/failureMatcher and deprecate the status based variant?

Reply:
I don't feel strongly about this either - moving away from 'magic' `Status` messages and specifying explicit protocol messages seems neat.

Reply:
We could add explicit messages for actorRef based streams (source and sink): https://github.com/nvollmar/akka/commit/b1c9e4dfed6765636cea61eb2386ec7d02022bf6

But I'd propose to open a separate issue for this since we'd need to adapt the API of all 4 variants (source/sink with/without ack).

Reply:
Created issue https://github.com/akka/akka/issues/27503